src/Microsoft.Identity.Web.Certificateless/ManagedIdentityClientAssertion.cs

```diff
@@ -39,9 +46,26 @@ public ManagedIdentityClientAssertion(string? managedIdentityClientId)
         /// <param name="managedIdentityClientId">Optional ClientId of the Managed Identity</param>
         /// <param name="tokenExchangeUrl">Optional audience of the token to be requested from Managed Identity. Default value is "api://AzureADTokenExchange". 
         /// This value is different on clouds other than Azure Public</param>
-        public ManagedIdentityClientAssertion(string? managedIdentityClientId, string? tokenExchangeUrl) : this (managedIdentityClientId)
+        /// <param name="logger">A logger</param>
+        public ManagedIdentityClientAssertion(string? managedIdentityClientId, string? tokenExchangeUrl, ILogger? logger)          
         {
             _tokenExchangeUrl = tokenExchangeUrl ?? CertificatelessConstants.DefaultTokenExchangeUrl;
+            _logger = logger;
+
+            var id = ManagedIdentityId.SystemAssigned;
+            if (!string.IsNullOrEmpty(managedIdentityClientId))
+            {
+                id = ManagedIdentityId.WithUserAssignedClientId(managedIdentityClientId);
+            }
+            var builder = ManagedIdentityApplicationBuilder.Create(id);
+            if (_logger != null)
+            {
+                builder.WithLogging(Log, ConvertMicrosoftExtensionsLogLevelToMsal(_logger));
```

Comment:
Added logging into MSAL for this.

Reply:
Don't we have this conversion already in IdWeb?

Reply:
We do, once for CCA, once for ManagedIdentityApplication. But I could not find a way to consolidate this code. 

I can move it all to the "Diagnostics" project, but in this case the Diagnostics project will depend on MSAL. Is this acceptable?

Reply:
would like to not have it in two places if possible, but also not in Diagnostics

Reply:
I'll just remove it for now.